main.go

```diff
@@ -133,6 +153,10 @@ func (c *powerDNSProviderSolver) Present(ch *v1alpha1.ChallengeRequest) error {
 		return fmt.Errorf("failed initializing powerdns provider: %v", err)
 	}
 
+	if !cfg.IsAllowedZone(ch.ResolvedZone) {
+		return fmt.Errorf("zone %s may not be edited per config", ch.ResolvedZone)
```

Comment:
Would it be beneficial to output the list of allowed zones in the error message? Something like:

```suggestion
		return fmt.Errorf("zone %s may not be edited per config: allowed zones %s", ch.ResolvedZone, cfg.AllowedZones)
```

Reply:
This could potentially be a quite long list, but OTOH, it will simplify late-night debugging significantly, so let's stick it in!